Adaptors/Redis/src/ObjectStorage.cs

```diff
@@ -155,8 +153,27 @@ await taskList.WhenAll()
                             .ToArray();
 
 
-    return await redis_.KeyDeleteAsync(keyList)
-                       .ConfigureAwait(false) == valuesCount + 1;
+    return await PerformActionWithRetry(() => redis_.KeyDeleteAsync(keyList)
+                       .ConfigureAwait(false)) == valuesCount + 1;
+  }
+
+  private T PerformActionWithRetry<T>(Func<T> action)
```

Comment:
```suggestion
  private async ValueTask<T> PerformActionWithRetry<T>(Func<ValueTask<T>> action)
```

I think you have issues with tasks here. You take a synchronous action that returns a task, so as soon as the task is created, your method will return without getting the chance to catch timeouts.

Reply:
indeed it was my problem, I updated to : await action() and now exception are catched

Reply:
can't use ValueTask, because I needt WaitAll and it is not available with ValueTask
Task.WhenAll(tasks.Select(t => t.AsTask()))
lead to memory allocation